proof/refine/AARCH64/ArchStateRelation.thy

```diff
@@ -289,133 +122,22 @@ definition is_other_obj_relation_type :: "a_type \<Rightarrow> bool" where
-  where
-  "ready_queue_relation d p q q' nexts prevs flag \<equiv>
-     list_queue_relation q q' nexts prevs
-     \<and> (\<forall>t. flag t \<longleftrightarrow> t \<in> set q)
-     \<and> (d > maxDomain \<or> p > maxPriority \<longrightarrow> tcbQueueEmpty q')"
-
-definition ready_queues_relation_2 ::
-  "(Structures_A.domain \<Rightarrow> Structures_A.priority \<Rightarrow> Structures_A.ready_queue)
-   \<Rightarrow> (domain \<times> priority \<Rightarrow> ready_queue)
-   \<Rightarrow> (obj_ref \<rightharpoonup> obj_ref) \<Rightarrow> (obj_ref \<rightharpoonup> obj_ref)
-   \<Rightarrow> (domain \<Rightarrow> priority \<Rightarrow> obj_ref \<Rightarrow> bool) \<Rightarrow> bool"
-  where
-  "ready_queues_relation_2 qs qs' nexts prevs inQs \<equiv>
-     \<forall>d p. let q = qs d p; q' = qs' (d, p); flag = inQs d p in
-           ready_queue_relation d p q q' nexts prevs flag"
-
-abbreviation ready_queues_relation :: "det_state \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "ready_queues_relation s s' \<equiv>
-     ready_queues_relation_2
-      (ready_queues s) (ksReadyQueues s') (tcbSchedNexts_of s') (tcbSchedPrevs_of s')
-      (\<lambda>d p. inQ d p |< tcbs_of' s')"
-
-lemmas ready_queues_relation_def = ready_queues_relation_2_def
-
 definition ghost_relation ::
   "Structures_A.kheap \<Rightarrow> (machine_word \<rightharpoonup> vmpage_size) \<Rightarrow> (machine_word \<rightharpoonup> nat) \<Rightarrow> (machine_word \<rightharpoonup> pt_type) \<Rightarrow> bool" where
   "ghost_relation h ups cns pt_types \<equiv>
      (\<forall>a sz. (\<exists>dev. h a = Some (ArchObj (DataPage dev sz))) \<longleftrightarrow> ups a = Some sz) \<and>
      (\<forall>a n. (\<exists>cs. h a = Some (CNode n cs) \<and> well_formed_cnode_n n cs) \<longleftrightarrow> cns a = Some n) \<and>
      (\<forall>a pt_t. (\<exists>pt. h a = Some (ArchObj (PageTable pt)) \<and> pt_t = pt_type pt) \<longleftrightarrow> pt_types a = Some pt_t)"
 
-definition cdt_relation :: "(cslot_ptr \<Rightarrow> bool) \<Rightarrow> cdt \<Rightarrow> cte_heap \<Rightarrow> bool" where
-  "cdt_relation \<equiv> \<lambda>cte_at m m'.
-     \<forall>c. cte_at c \<longrightarrow> cte_map ` descendants_of c m = descendants_of' (cte_map c) m'"
-
-definition cdt_list_relation :: "cdt_list \<Rightarrow> cdt \<Rightarrow> cte_heap \<Rightarrow> bool" where
- "cdt_list_relation \<equiv> \<lambda>t m m'.
-    \<forall>c cap node. m' (cte_map c) = Some (CTE cap node)
-        \<longrightarrow> (case next_slot c t m of None \<Rightarrow> True
-                                   | Some next \<Rightarrow> mdbNext node = cte_map next)"
+(* FIXME arch-split: provided only so that ghost_relation can be used within generic definition
+   of state_relation (since arch_state_relation doesn't have access to kheap, and
+   gsPTTypes on AARCH64 isn't generic) *)
+definition ghost_relation_wrapper :: "det_state \<Rightarrow> kernel_state \<Rightarrow> bool" where
```

Comment:
do we want it to be `det_state`? it looks like it would be fine the generic state

Reply:
Turns out on master all `state_relation` takes `det_state` already, and that's where I got it from. After discussion, we both decided that indicates no need for generalisation here.